### PR TITLE
feat(ui5-message-strip): introduce custom colors

### PR DESCRIPTION
**Pull request by niyap**
With the following enhancement, the end user will be able to create message strips with custom colors from a predefined set of colors. That is achieved enhancing the `design` property enum `MessageStripDesign` with two new options `ColorSet1` and `ColorSet2` as well as the newly introduced property `colorScheme`.
Setting the `design` property value to `ColorSet1` or `ColorSet2`, then you are able to choose between 10 different colors for each using `colorScheme` property with values from "1" to "10".

### CODE REVIEW

File: packages/main/src/MessageStrip.ts [hunk 1]
```diff
@@ -35,13 +36,8 @@ enum DesignClassesMapping {
...
 	Negative = "ui5-message-strip-root--negative",
 	Warning = "ui5-message-strip-root--warning",
-}
-
-enum IconMapping {
-	Information = "information",
-	Positive = "sys-enter-2",
-	Negative = "error",
-	Warning = "alert",
+	ColorSet1 = "ui5-message-strip-root--color-set-1",
```
Comment (elenastoyanovaa) [inline packages/main/src/MessageStrip.ts:39]: is it discussed to have default icons?

File: packages/main/src/themes/base/MessageStrip-parameters.css [hunk 2]
```diff
@@ -15,4 +15,29 @@
 	--_ui5_message_strip_focus_offset: -2px;
 	--_ui5_message_strip_close_button_top: 0.125rem;
 	--_ui5_message_strip_close_button_right: 0.125rem;
+	/* Next to variables below are hardcoded values of sapButton_Lite_Hover_Background parameter with specified opacity*/
```
Comment (elenastoyanovaa) [inline packages/main/src/themes/base/MessageStrip-parameters.css:18]: two

File: packages/main/src/themes/sap_fiori_3/MessageStrip-parameters.css [hunk 3]
```diff
@@ -0,0 +1,7 @@
+@import "../base/MessageStrip-parameters.css";
+
+:root{
+	/* Next to variables below are hardcoded values of sapButton_Lite_Hover_Background parameter with specified opacity*/
```
Comment (elenastoyanovaa) [inline packages/main/src/themes/sap_fiori_3/MessageStrip-parameters.css:4]: two

File: packages/main/src/themes/sap_fiori_3_dark/MessageStrip-parameters.css [hunk 4]
```diff
@@ -0,0 +1,9 @@
+@import "../base/MessageStrip-parameters.css";
+
+:root{
+	/* Next to variables below are hardcoded values of sapButton_Lite_Hover_Background parameter with specified opacity*/
```
Comment (elenastoyanovaa) [inline packages/main/src/themes/sap_fiori_3_dark/MessageStrip-parameters.css:4]: two

File: packages/main/src/themes/sap_horizon_dark/MessageStrip-parameters.css [hunk 5]
```diff
@@ -2,4 +2,9 @@
 
 :root{
 	--_ui5_message_strip_close_button_right: 0.1875rem;
+	/* Next to variables below are hardcoded values of sapButton_Lite_Hover_Background parameter with specified opacity*/
```
Comment (elenastoyanovaa) [inline packages/main/src/themes/sap_horizon_dark/MessageStrip-parameters.css:5]: two

File: packages/main/src/types/MessageStripDesign.ts [hunk 6]
```diff
@@ -26,6 +26,18 @@ enum MessageStripDesign {
 	 * @public
 	 */
 	Warning = "Warning",
+
+	/**
+	* Message is custom
```
Comment (elenastoyanovaa) [inline packages/main/src/types/MessageStripDesign.ts:31]: Maybe change it to - Message uses custom Color Set 1, or something which differentiates between the two enum values

File: packages/main/test/pages/MessageStrip.html [hunk 7]
```diff
@@ -39,9 +39,28 @@
 	<ui5-message-strip hide-icon hide-close-button class="top" design="Information">Ea mollit nulla laborum et fugiat nulla excepteur ea. Duis et dolor enim Lorem laboris adipisicing cillum quis proident dolor veniam voluptate. Nostrud dolore ipsum anim voluptate enim dolore eiusmod aliqua et. Est eu ex dolor ea ipsum. Adipisicing duis aliquip ullamco culpa dolore exercitation ullamco cillum irure.</ui5-message-strip>
 	<ui5-message-strip hide-close-button class="top" design="Information">Ea mollit nulla laborum et fugiat nulla excepteur ea. Duis et dolor enim Lorem laboris adipisicing cillum quis proident dolor veniam voluptate. Nostrud dolore ipsum anim voluptate enim dolore eiusmod aliqua et. Est eu ex dolor ea ipsum. Adipisicing duis aliquip ullamco culpa dolore exercitation ullamco cillum irure.</ui5-message-strip>
 
-    <ui5-message-strip design="Warning" class="top"><ui5-icon name="palette" slot="icon"></ui5-icon>Custom icon</ui5-message-strip>
```
Comment (elenastoyanovaa) [inline packages/main/test/pages/MessageStrip.html:42]: let's not delete those two examples

**Review by elenastoyanovaa — CHANGES_REQUESTED** [review]
+ Add a sample to the new playground

**Review by elenastoyanovaa — COMMENTED** [review]
Regarding the commit message - maybe custom messages is not the best description. The customisation is actually only visual. Maybe something like 

> With the following enhancement, the end user will be able to message strips with custom colors, from a predefined set of colors.

File: packages/main/src/MessageStrip.ts [hunk 8]
```diff
@@ -108,6 +104,17 @@ class MessageStrip extends UI5Element {
...
+	 *
+	 * @default "1"
+	 * @public
+	 */
+	@property({ defaultValue: "1" })
+	colorScheme!: string;
+
 	/**
 	 * Defines whether the MessageStrip will show an icon in the beginning.
 	 * You can directly provide an icon with the `icon` slot. Otherwise, the default icon for the type will be used.
```
Comment (elenastoyanovaa) [inline packages/main/src/MessageStrip.ts:120]: I would add a note in the hideIcon property or/and in the icon slot docu stating that:

> Keep in mind that if a `ColorSet1` and `ColorSet2` design is used, not default icon is present. Applications are responsible to provide a slotted icon or to apply the hideIcon property.

**Review by elenastoyanovaa — CHANGES_REQUESTED** [review]

File: packages/main/src/MessageStrip.ts [hunk 9]
```diff
@@ -108,9 +104,23 @@ class MessageStrip extends UI5Element {
...
+	 *
+	 *  * **Note:** If <code>MessageStripDesign.ColorSet1</code> or <code>MessageStripDesign.ColorSet2</code> value is set to the <code>design</code> property, default icon will not be presented.
```
Comment (simonarangelova) [inline packages/main/src/MessageStrip.ts:122]:
```suggestion
	 is set to the <code>design</code> property, the default icon will not be presented.
```

**Comment by niyap** [comment]
> Regarding the commit message - maybe custom messages is not the best description. The customisation is actually only visual. Maybe something like
> 
> > With the following enhancement, the end user will be able to message strips with custom colors, from a predefined set of colors.

Done

File: packages/main/src/MessageStrip.ts (hunk 8 above)
Comment (niyap) [reply packages/main/src/MessageStrip.ts:120]: Done

**Review by simonarangelova — APPROVED** [review]

File: packages/main/src/MessageStrip.ts [hunk 10]
```diff
@@ -108,9 +104,23 @@ class MessageStrip extends UI5Element {
...
+	/**
+	 * Defines the color scheme of the component.
+	 * There are 10 predefined schemes.
+	 * To use one you can set a number from `"1"` to `"10"`. The `colorScheme` `"1"` will be set by default.
+	 *
+	 * @default "1"
+	 * @public
+	 */
+	@property({ defaultValue: "1" })
+	colorScheme!: string;
```
Comment (elenastoyanovaa) [inline packages/main/src/MessageStrip.ts:116]: add since tag

**Review by elenastoyanovaa — CHANGES_REQUESTED** [review]

File: packages/main/src/MessageStrip.ts (hunk 10 above)
Comment (niyap) [reply packages/main/src/MessageStrip.ts:117]: Done

**Pull request merged**